hack/verify-makegen.sh

```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+result=$(git diff ${OS_OUTPUT_BINPATH})
```

Comment:
This script should do more than just check for a git diff, it should ideally actually run the generator scripts in a temp directory and compare that to the current output. The reason for this is because this script on its own isn't verifying `make gen`, so if I call it directly with `./hack/verify-makegen.sh` or `make verify-gen`, I won't really be checking that.

You can see how [`verify-vendor.sh`](https://github.com/kubernetes-sigs/descheduler/blob/6ecbc85/hack/verify-vendor.sh#L52-L58) works for an example.

Following that pattern, I think it would be better to have separate `./hack/verify-xx...` scripts for each of the generators (for the same reason: that I may want to just verify one of them).

Reply:
Hi @damemi thanks for pointing out the reference script `verify-vendor.sh`. I wanted to check, is this the line which suggests correct way of using `git diff` : [L64](https://github.com/kubernetes-sigs/descheduler/blob/6ecbc85/hack/verify-vendor.sh#L64) ? Also, in our case for the `git diff` to compare, do we need to take output generated by `./hack/updated-generated-xxx` scripts in the latest commit with the output generated in the `temp_` directory( created during PR check) ?

Reply:
I was referring more to how this script is pushing the existing changes onto a tmp directory, running the generator, and then comparing if any new changes occurred. So yes, the diff should be comparing the latest commit to a copy of the latest commit that has had the generator run. In this case, you are just getting that by doing `git diff` on the copy